lib/socks.c

```diff
@@ -507,7 +503,7 @@ CURLproxycode Curl_SOCKS5(const char *proxy_user,
     o  X'00' succeeded
   */
   struct connectdata *conn = data->conn;
-  unsigned char *socksreq = &conn->cnnct.socksreq[0];
+  unsigned char *socksreq = (unsigned char *)data->state.buffer;
```

Comment:
It's probably worth having an assert on the minimum buffer length here in case somebody ever tries to override the 1000 byte minimum.

Reply:
Sound advice!